js/snippetPreviewToggler.js

```diff
@@ -0,0 +1,153 @@
+ *
+ * @param {string}    previewMode    The default preview mode.
+ * @param {Element[]} previewToggles Array with toggle elements.
+ *
+ * @property {string}   previewMode    The current preview mode.
+ * @propert {Element[]} previewToggles The array with toggle elements.
+ * @constructor
+ */
+var SnippetPreviewToggler = function( previewMode, previewToggles ) {
+	this.previewMode    = previewMode;
+	this.previewToggles = previewToggles;
+};
+
+/**
+ * Initializes the object by setting the current previewMode as the active one.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.initialize = function() {
+	this._setPreviewMode( this.previewMode, this._findElementByMode( this.previewMode ) );
+};
+
+/**
+ * Sets the events for the preview toggles to switch between the preview modes.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.setEvents = function() {
+	forEach( this.previewToggles, function( previewToggle ) {
+		previewToggle.addEventListener( "click", function( evt ) {
+			evt.preventDefault();
+
+			this._setPreviewMode( previewToggle.getAttribute( "data-type" ), previewToggle );
+		}.bind( this ) );
+	}.bind( this ) );
+};
+
+/**
+ * Returns the element by given node.
+ *
+ * @param {string} previewMode The element to search.
+ * @returns {Element} The found element.
+ * @private
+ */
+SnippetPreviewToggler.prototype._findElementByMode = function( previewMode ) {
+	return document.getElementsByClassName( "snippet-editor__view-icon-" + previewMode )[ 0 ];
+};
+
+/**
+ * Sets the preview mode.
+ *
+ * @param {string}  previewMode   The preview mode that has to be set.
+ * @param {Element} toggleElement The element that has been triggered.
+ *
+ * @returns {void}
+ * @private
+ */
+SnippetPreviewToggler.prototype._setPreviewMode = function( previewMode, toggleElement ) {
+	this._removeActiveStates();
+	this._setActiveState( toggleElement );
+
+	if( previewMode === this.previewMode ) {
+		return;
+	}
+
+	var viewElement = document.getElementById( "snippet-preview-view" );
+
+	domManipulation.removeClass( viewElement, previewModes[ this.previewMode ] );
+	domManipulation.addClass( viewElement, previewModes[ previewMode ] );
+
+	this.previewMode = previewMode;
+};
+
+/**
+ * Sets the Snippet Preview Toggler to desktop mode.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.setDesktopMode = function() {
+	this._setPreviewMode( "desktop", this._findElementByMode( "desktop" ) );
+};
+
+/**
+ * Sets the Snippet Preview Toggler to desktop mode.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.setMobileMode = function() {
+	this._setPreviewMode( "mobile", this._findElementByMode( "mobile" ) );
+};
+
+/**
+ * Sets the visibility based on the width of the Snippet Preview container.
+ *
+ * @param {number} previewWidth the width of the Snippet Preview container.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.setToggleVisibility = function( previewWidth ) {
```

Comment:
`setToggleVisibility` seems like a strange name. Aren't you just toggling the visibility? I'd consider changing this to either `toggleVisibility` or `setVisibility`. Please don't forget to update the documentation if you decide to change this name.